File: leetcode-04/solutions/python3/0069.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l = 0
        r = x + 1

        while l < r:
            m = (l + r) >> 1
            if m * m > x:
                r = m
            else:
                l = m + 1

        return l - 1

File: leetcode-04/solutions/cpp/0069.cpp
class Solution {
 public:
  int mySqrt(int x) {
    unsigned l = 1;
    unsigned r = x + 1u;

    while (l < r) {
      unsigned m = (l + r) >> 1;
      if (m > x / m)
        r = m;
      else
        l = m + 1;
    }

    return l - 1;
  }
};